src/xo-app/new-vm/index.js

```diff
@@ -995,53 +997,70 @@ export default class NewVm extends BaseComponent {
     const { state: { VIFs } } = this.state
     const { pool } = this.props
     const { formatMessage } = this.props.intl
+    const resourceSet = this._getResolvedResourceSet()
 
     return <Section icon='new-vm-interfaces' title='newVmInterfacesPanel' done={this._isInterfacesDone()}>
       <SectionContent column>
-        {map(VIFs, (vif, index) => <div key={index}>
-          <LineItem>
-            <Item label={_('newVmMacLabel')}>
-              <DebounceInput
-                className='form-control'
-                debounceTimeout={DEBOUNCE_TIMEOUT}
-                onChange={this._linkState(`VIFs.${index}.mac`)}
-                placeholder={formatMessage(messages.newVmMacPlaceholder)}
-                rows={7}
-                value={vif.mac}
-              />
-            </Item>
-            <Item label={_('newVmNetworkLabel')}>
-              <span className={styles.inlineSelect}>
-                {pool ? <SelectNetwork
-                  onChange={this._linkState(`VIFs.${index}.network`, 'id')}
-                  predicate={this._getNetworkPredicate()}
-                  value={vif.network}
-                />
-                : <SelectResourceSetsNetwork
-                  onChange={this._linkState(`VIFs.${index}.network`, 'id')}
-                  resourceSet={this._getResolvedResourceSet()}
-                  value={vif.network}
-                />}
-              </span>
-            </Item>
+        {map(VIFs, (vif, index) => {
+          const ipPoolPredicate = ipPool => {
```

Comment:
The predicate is regenerated at each render which triggers IP addresses filterings.

You should isolate each VIF in its own component.